### PR TITLE
Hotfix wallet dartindex

### PR DESCRIPTION
**Pull request by imrying**
This changes the format from bool[Pubkey] and TagionBill[Pubkey] to bool[DARTIndex] and TagionBill[DARTIndex] in order to support multiple bills on the same public key.

This is a breaking change and will cause old wallets to break on updates.

### CODE REVIEW

**Comment by lucasnethaj** [comment]
This is not backwards compatible correct?

**Comment by imrying** [comment]
> This is not backwards compatible correct?

No, this is a breaking update.

**Comment by lucasnethaj** [comment]
> > This is not backwards compatible correct?
> 
> No, this is a breaking update.

Ok because obviously it's not good to make a breaking change in non mayor update. Are there any existing wallets that would need to be upgraded currently

File: src/lib-wallet/tagion/wallet/AccountDetails.d [hunk 1]
```diff
@@ -123,37 +113,29 @@ struct AccountDetails {
...
     }
     /++
          Clear up the Account
          Remove used bills
          +/
-    void clearup() pure {
+    void clearup() {
         bills
             .filter!(b => b.owner in derivers)
             .each!(b => derivers.remove(b.owner));
```
Comment (cbleser) [inline src/lib-wallet/tagion/wallet/AccountDetails.d:125]: It is a problem if we remove the derivers if we have more than one bill with the same public key

File: src/lib-wallet/tagion/wallet/SecureWallet.d [hunk 2]
```diff
@@ -619,27 +620,23 @@ struct SecureWallet(Net : SecureNet) {
...
             .filter!(a => a.value == true)
             .map!(a => a.key)
             .array;
 
-        auto found_owners = found_bills.map!(found => found.owner).array;
-        foreach (pkey; locked_pkeys) {
-            if (!(found_owners.canFind(pkey))) {
-                account.activated.remove(pkey);
-                auto bill_index = account.bills.countUntil!(b => b.owner == pkey);
+        auto found_indexes = found_bills.map!(found => net.dartIndex(found)).array;
```
Comment (cbleser) [inline src/lib-wallet/tagion/wallet/SecureWallet.d:629]: we use the spelling indices elsewhere

**Review by cbleser — COMMENTED** [review]
Done

File: src/lib-wallet/tagion/wallet/AccountDetails.d (hunk 1 above)
Comment (imrying) [reply src/lib-wallet/tagion/wallet/AccountDetails.d:126]: function is unused. WIll version(none) for now.

**Comment by imrying** [comment]
> > > This is not backwards compatible correct?
> > 
> > 
> > No, this is a breaking update.
> 
> Ok because obviously it's not good to make a breaking change in non mayor update. Are there any existing wallets that would need to be upgraded currently

Yes auzahlung and one wallet were a test payment was performed.

**Review by lucasnethaj — APPROVED** [review]

**Review by cbleser — APPROVED** [review]

**Pull request merged**